holidays/countries/senegal.py

```diff
@@ -0,0 +1,193 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.calendars.gregorian import MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, DEC
+from holidays.groups import (
+    ChristianHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+    StaticHolidays,
+)
+from holidays.observed_holiday_base import ObservedHolidayBase, SUN_TO_NEXT_MON
+
+
+class Senegal(
+    ObservedHolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays, StaticHolidays
+):
+    """Senegal holidays.
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_Senegal>
+        * [Law 63-51](https://web.archive.org/web/20250604134724/https://www.dri.gouv.sn/sites/default/files/LOI/1963/63_51.pdf)
+        * [Law 74-52](https://web.archive.org/web/20250608105243/https://www.dri.gouv.sn/sites/default/files/an-documents/LOI%20N1974%2052%20DU%204%20NOVEMBRE%201974.pdf)
+        * [Law 83-54](https://web.archive.org/web/20250608105419/https://www.dri.gouv.sn/sites/default/files/LOI/1983/comP4%20loi%20decentralisation%20et%20travail/LOI%20N%20198354%20DU%2018%20FEVRIER%201983/LOI%20N%20198354%20DU%2018%20FEVRIER%201983.pdf)
+        * [Law 2013-06](https://web.archive.org/web/20250604215148/https://natlex.ilo.org/dyn/natlex2/natlex2/files/download/97261/SEN-97261.pdf)
+    """
+
+    country = "SN"
+    default_language = "fr_SN"
+    # %s (estimated).
+    estimated_label = tr("%s (estimé)")
+    # %s (observed).
+    observed_label = tr("%s (observé)")
+    # %s (observed, estimated).
+    observed_estimated_label = tr("%s (observé, estimé)")
+    # First official source with holidays dates (Law 63-51).
+    start_year = 1964
+    supported_languages = ("en_US", "fr_SN")
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
```

Comment:
Last minute changes in favour of PR #2618 refactor, although not necessarily required as we can simply merge this PR first.
```suggestion
    def __init__(self, *args, islamic_show_estimated: bool = True, **kwargs):
```